File: shelf_space_allocation/shelf_space_allocation_starter.py
## Shelf space allocation starter
import pandas as pd

p = pd.read_csv("products_set_30.csv")
print(p)

# derived metrics
p['monthly_profit'] = p['unit_margin'] * p['monthly_demand']
p['monthly_profit_per_width'] = p['monthly_profit'] / p['width']
print(p)

# top products by monthly profit
print(p.sort_values('monthly_profit', ascending=False).head(10))

# top products by monthly profit per unit width
print(p.sort_values('monthly_profit_per_width', ascending=False).head(10))

# brands
print(p.groupby('brand_id').count().iloc[:,1])

# total width of all products
print(p['width'].sum())

### suggestion: add a column of 1/0 for product facings
### track total revenue, total width, products per brand

File: shelf_space_allocation/shelf_space_allocation_starter.R
#' ---
#' title: "Shelf space allocation starter"
#' author: "Pavel Paramonov"
#' date: "Retail Analytics (2021)"
#' ---
library(tidyverse)
library(DT)

# product list
p0 <- read_csv("products_set_30.csv")
p0

# derived metrics
p <- p0 %>% mutate(monthly_profit = unit_margin * monthly_demand, .before = "category_id") %>%
  mutate(monthly_profit_per_width = unit_margin * monthly_demand / width, .before = "category_id")

# top products by monthly profit
p %>% slice_max(monthly_profit, n=10)

# top products by monthly profit per unit width
p %>% slice_max(monthly_profit_per_width, n=10)

# full set of products
datatable(p)

# brands
p %>% count(brand_id)

# total width of all products
p %>% summarize(width_total = sum(width))

### suggestion: add a column of 1/0 for product facings
### track total revenue, total width, products per brand
